cli/cmd/component-delete.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/lokomotive/pkg/components"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+	"github.com/kinvolk/lokomotive/pkg/config"
+)
+
+var componentDeleteCmd = &cobra.Command{
+	Use:   "delete",
+	Short: `Delete a component installed in the cluster.`,
+	Run:   runDelete,
+}
+
+var deleteNamespace bool
+
+// nolint:gochecknoinits
+func init() {
+	componentCmd.AddCommand(componentDeleteCmd)
+	pf := componentDeleteCmd.PersistentFlags()
+	pf.BoolVarP(&deleteNamespace, "delete-namespace", "", false, "Delete namespace with component.")
+}
+
+func runDelete(cmd *cobra.Command, args []string) {
+	contextLogger := log.WithFields(log.Fields{
+		"command": "lokoctl component delete",
+		"args":    args,
+	})
+
+	lokoCfg, diags := getLokoConfig()
+	if len(diags) > 0 {
+		contextLogger.Fatal(diags)
+	}
+
+	componentsToDelete := make([]string, len(args))
+	copy(componentsToDelete, args)
+
+	if len(args) == 0 {
+		componentsToDelete = make([]string, len(lokoCfg.RootConfig.Components))
+
+		for i, component := range lokoCfg.RootConfig.Components {
+			componentsToDelete[i] = component.Name
+		}
+	}
+
+	if !askForConfirmation(
+		fmt.Sprintf("Do you want to delete components: %q?", strings.Join(componentsToDelete, ",")),
```

Comment:
I think that when more than 3 components are to be deleted, it's not very readable to have them in the `"component1,component2,component3"` format.
It be nice if they were instead shown one per line:
```
The following components will be deleted:
  component1
  component2
  component3
  component4

Are you sure you want to proceed? [type Yes to continue and delete the components]
```

Reply:
```console
$ lokoctl component delete --delete-namespace
The following components will be deleted:
openebs-operator
        openebs-storage-class
        prometheus-operator
        contour
        metallb
        cert-manager
        httpbin
        rook
        rook-ceph
        flatcar-linux-update-operator
        metrics-server
        external-dns
        cluster-autoscaler
        dex
        gangway

Are you sure you want to proceed? [type "yes" to continue]: no
INFO[0003] Components deletion cancelled.                args="[]" command="lokoctl component delete"

```

changed the code and now it looks like above.

Reply:
It seem the line with `openebs-operator` is wrongly formatted

Reply:
oh yeah missed that.

```console
$ lokoctl component delete --delete-namespace
The following components will be deleted:
        openebs-operator
        openebs-storage-class
        prometheus-operator
        contour
        metallb
        cert-manager
        httpbin
        rook
        rook-ceph
        flatcar-linux-update-operator
        metrics-server
        external-dns
        cluster-autoscaler
        dex
        gangway

Are you sure you want to proceed? [type "yes" to continue]: no
INFO[0002] Components deletion cancelled.                args="[]" command="lokoctl component delete"
```